backend/app/services/rule_monitor.py

```
@@ -0,0 +1,381 @@
+    """
+    captured_pct = (profit_target_status or {}).get("captured_pct")
+    is_itm = moneyness_state == "ITM"
+    in_warning_window = 0 <= dte <= expiration_warning_days
+    in_dte_window = 0 <= dte <= dte_review_days
+    expired = dte < 0
+
+    # --- assignment (7d ITM) — never a countdown: triggered or no. ----------
+    if in_warning_window and is_itm:
+        assignment_status = "triggered"
+    else:
+        assignment_status = "no"
```

Ruff (0.15.13)

[warning] Use ternary operator `assignment_status = "triggered" if in_warning_window and is_itm else "no"` instead of `if`-`else`-block

Replace `if`-`else`-block with `assignment_status = "triggered" if in_warning_window and is_itm else "no"`

(SIM108)